src/Controller.ts

```diff
@@ -39,41 +39,38 @@ export class Controller {
     public async startBootApp(app: BootApp, debug?: boolean): Promise<void> {
         const mainClasData = await vscode.window.withProgress(
             { location: vscode.ProgressLocation.Window, title: `Resolving main classes for ${app.name}...` },
-            () => { return this._getMainClass(app.path); }
+            () => { return this._getMainClass(app); }
         );
+        if (!mainClasData) {
+            vscode.window.showWarningMessage(mainClasData === null ?
```

Comment:
should not show warning if user cancels the selection.

Reply:
agree.